integration-tests/test_registration.py

```diff
@@ -1,22 +1,91 @@
+    """
+    response = insights_client.run("--register", check=False)
+    assert (
+        response.returncode == 0
+    ), "An application returns 0 after successful registration"
+    assert (
+        "Successfully registered host" in response.stdout
+    ), "An application should inform a user that a system was registered"
+    assert (
+        "Starting to collect Insights data" in response.stdout
+    ), "An application should inform a user that Insights data will be collected"
+    assert (
+        "Successfully uploaded report" in response.stdout
+    ), "An application should inform a user that report was successfully uploaded"
+    assert (
+        "View the Red Hat Insights console" in response.stdout
+    ), "An application should inform a user about Insights console"
+    assert (
+        insights_client.is_registered
+    ), "Current state of registration should be 'registered'"
+
+
+def test_double_register(insights_client):
+    """`--register` can be passed multiple times.
+
+    Even system that is already registered should allow the `--register` flag to be
+    passed in, without resulting in non-zero exit code.
+
+    This behavior has changed multiple times during the package lifetime.
+    """
+    assert not insights_client.is_registered
+
+    insights_client.register()
+    assert os.path.exists(
+        MACHINE_ID_FILE
+    ), f"{MACHINE_ID_FILE} should exist after successfull registration"
+
+    with open(MACHINE_ID_FILE, "r") as f:
+        machine_id_old = f.read()
+
+    response = insights_client.register()
+    assert (
+        "This host has already been registered" in response.stdout
+    ), "An application should inform that a system was already registered"
+    assert (
+        "Successfully uploaded report" in response.stdout
+    ), "An application should inform that report was successfully uploaded"
+    assert (
+        "View the Red Hat Insights console" in response.stdout
+    ), "An application should inform about Insights console"
+    assert (
+        insights_client.is_registered
+    ), "Current state of registration should be 'registered'"
+
+    assert os.path.exists(MACHINE_ID_FILE)
+
+    with open(MACHINE_ID_FILE, "r") as f:
+        machine_id_new = f.read()
+
+    assert (
+        machine_id_new == machine_id_old
+    ), f"{MACHINE_ID_FILE} should remain the same after repeated registration"
+
+
 def test_machineid_exists_only_when_registered(insights_client):
     """`machine-id` is only present when insights-client is registered."""
     assert not insights_client.is_registered
     assert not os.path.exists(MACHINE_ID_FILE)
 
     res = insights_client.run(check=False)
     assert (
-        "This host has not been registered. Use --register to register this host."
-    ) in res.stdout
+        "This machine has not yet been registered. Use --register to register"
+    ) in res.stdout, "An application should ask a user to register a system"
```

Comment:
the message I get when testing actually mentions "host", so I'm not sure why this assert was changed (the commit message does not mention anything about it)